net-libs/libcec/patches/libcec-4.0.3.patchset

```diff
@@ -0,0 +1,132 @@
++                        ${CMAKE_INSTALL_INCLUDEDIR})
+ 
+   install(FILES         ${CMAKE_CURRENT_SOURCE_DIR}/libcec.pc
+           DESTINATION   ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+@@ -176,7 +176,7 @@ install(FILES ${PROJECT_SOURCE_DIR}/../../include/cec.h
+               ${PROJECT_SOURCE_DIR}/../../include/ceccloader.h
+               ${PROJECT_SOURCE_DIR}/../../include/cectypes.h
+               ${PROJECT_SOURCE_DIR}/../../include/version.h
+-        DESTINATION include/libcec)
++        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libcec)
+ 
+ # libCEC target
+ add_library(cec SHARED ${CEC_SOURCES})
+-- 
+2.20.1
+
+
+From bf4468f368b3af1f1676f9509c4f730343c9d79b Mon Sep 17 00:00:00 2001
+From: TURX <turx2003@gmail.com>
+Date: Thu, 23 Jan 2020 17:23:19 +0800
+Subject: [PATCH] fix Haiku support
+
+---
+ src/cec-client/CMakeLists.txt            | 12 ++++++++++--
+ src/libcec/cmake/SetBuildInfo.cmake      |  4 ++++
+ src/libcec/platform/drm/drm-edid.cpp     |  2 ++
+ src/libcec/platform/posix/serialport.cpp |  4 ++++
+ 4 files changed, 20 insertions(+), 2 deletions(-)
+
+diff --git a/src/cec-client/CMakeLists.txt b/src/cec-client/CMakeLists.txt
+index 8002d49..30fa489 100644
+--- a/src/cec-client/CMakeLists.txt
++++ b/src/cec-client/CMakeLists.txt
+@@ -24,7 +24,11 @@ find_package(Threads REQUIRED)
+ set(cecclient_SOURCES cec-client.cpp)
+ 
+ # curses
+-check_library_exists(curses initscr "" HAVE_CURSES_API)
++if (HAIKU)
++  check_library_exists(ncurses initscr "" HAVE_CURSES_API)
++else()
++  check_library_exists(curses initscr "" HAVE_CURSES_API)
++endif()
+ if (HAVE_CURSES_API)
+   list(APPEND cecclient_SOURCES curses/CursesControl.cpp)
+ 
+@@ -47,7 +51,11 @@ if (NOT WIN32)
+ 
+   # curses
+   if (HAVE_CURSES_API)
+-    target_link_libraries(cec-client curses)
++    if (HAIKU)
++      target_link_libraries(cec-client ncurses)
++    else()
++      target_link_libraries(cec-client curses)
++    endif()
+     if (HAVE_CURSES_TINFO)
+       target_link_libraries(cec-client tinfo)
+     endif()
+diff --git a/src/libcec/cmake/SetBuildInfo.cmake b/src/libcec/cmake/SetBuildInfo.cmake
+index 39bf4be..f0859cd 100644
+--- a/src/libcec/cmake/SetBuildInfo.cmake
++++ b/src/libcec/cmake/SetBuildInfo.cmake
+@@ -45,7 +45,11 @@ else()
+   # add host on which this was built to compile info
+   find_program(HAVE_HOSTNAME_BIN hostname /bin /usr/bin /usr/local/bin)
+   if(HAVE_HOSTNAME_BIN)
++    if(HAIKU)
++    exec_program(hostname OUTPUT_VARIABLE BUILD_HOST)
++    else()
+     exec_program(hostname ARGS -f OUTPUT_VARIABLE BUILD_HOST)
++    endif()
+     set(LIB_INFO "${LIB_INFO}@${BUILD_HOST}")
+   endif()
+ 
+diff --git a/src/libcec/platform/drm/drm-edid.cpp b/src/libcec/platform/drm/drm-edid.cpp
+index a3b1596..2a094d1 100644
+--- a/src/libcec/platform/drm/drm-edid.cpp
++++ b/src/libcec/platform/drm/drm-edid.cpp
+@@ -61,6 +61,7 @@ uint16_t CDRMEdidParser::GetPhysicalAddress(void)
+ 
+   while (entry != NULL && enablededid.empty())
+   {
++#ifndef __HAIKU__
```

Comment:
@korli should we comment the whole block out? it's failing on d_type

Reply:
This code makes no sense, we don't have a /sys/.. file tree.